plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-shipping-methods-v2-controller.php

```diff
@@ -101,14 +101,19 @@ public function get_item_permissions_check( $request ) {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_items( $request ) {
+		
 		$wc_shipping = WC_Shipping::instance();
-		$response    = array();
+		$data        = array();
 		foreach ( $wc_shipping->get_shipping_methods() as $id => $shipping_method ) {
-			$method     = $this->prepare_item_for_response( $shipping_method, $request );
-			$method     = $this->prepare_response_for_collection( $method );
-			$response[] = $method;
+			$method = $this->prepare_item_for_response( $shipping_method, $request );
+			$method = $this->prepare_response_for_collection( $method );
+			$data[] = $method;
 		}
-		return rest_ensure_response( $response );
+
+		$total    = count( $data );
+		$response = rest_ensure_response( $data );
+		$response->header( 'X-WP-Total', $total );
```

Comment:
It'd be a good opportunity to [also set `X-WP-TotalPages`](https://developer.wordpress.org/rest-api/using-the-rest-api/pagination/) for this and the other requests (obviously set to `1`). Even though this isn't used by the WP-CLI tool, it could be useful to proper REST requests.

Reply:
Done